### PR TITLE
Document Reasons for missing Preconditions

### PR DESCRIPTION
**Pull request by ben-Draeger**
Added either a Note explaining the reason or a TODO for adding a Precondition to Invariant Test Cases that do not have a Precondition, although they seem to need one.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Adherence to coding conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Adherence to javadoc conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* config update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* SDCcc executable ran against a test device (if necessary)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java [hunk 1]
```diff
@@ -298,6 +298,7 @@ void testRequirementR001200() throws NoTestData, IOException {
         assertTestData(acceptableSequenceSeen.get(), "No AlertConditionState seen during the test run, test failed.");
     }
 
+    // TODO: Add a Precondition (as part of SDCCC-1348)
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java:301]: Do not reference internal issues.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java [hunk 2]
```diff
@@ -341,6 +342,7 @@ void testRequirementR0013() throws NoTestData, IOException {
         assertTestData(acceptableSequenceSeen.get(), "No suitable context states seen, test failed.");
     }
 
+    // TODO: Add a Precondition (as part of SDCCC-1348)
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java:345]: Do not reference internal issues.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java [hunk 3]
```diff
@@ -42,6 +42,10 @@ void setUp() {
         this.messageStorage = injector.getInstance(MessageStorage.class);
     }
 
+    // NOTE: No PreCondition can ensure that there are Localized texts with a @Ref attribute. This is because
+    //       a device either provides a LocalizationService, in which case all Localized texts will have a @Ref attribute,
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java:46]: This is not true, most texts will have a `@Ref`, but there is still, e.g., user-entered text and also the SystemErrorReport, in which the `@Ref` is explicitly forbidden.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java [hunk 4]
```diff
@@ -42,6 +42,10 @@ void setUp() {
...
+    //       or it doesn't, in which case no Localized text will have a @Ref attribute.
+    //       When a device does not provide a LocalizationService, then this test should not be applied to it.
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java:48]: This is definitely not the right place to "inform" anyone when not to apply the test, no user will ever read it.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java [hunk 5]
```diff
@@ -121,6 +121,9 @@ void testRequirementB6() throws NoTestData, IOException {
                 acceptableSequenceSeen.get(), "No AbstractOperationDescriptors seen during test run, test failed.");
     }
 
+    // NOTE: No PreCondition can ensure that an ApprovedJurisdictions list is present or absent as this
+    //       list is supposed to be static for a Mds. When all Mds in a Device's Mdib have an ApprovedJurisdictions
+    //       list, then this test case should not be applied to it.
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java:126]: This is definitely not the right place to "inform" anyone when not to apply the test, no user will ever read it.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java [hunk 6]
```diff
@@ -166,6 +169,9 @@ void testRequirementB284() throws NoTestData, IOException {
         assertTestData(acceptableSequenceSeen.get(), "No acceptable sequence seen, test failed.");
     }
 
+    // NOTE: No PreCondition can ensure that an ApprovedJurisdictions list is present or absent as this
+    //       list is supposed to be static for a Vmd. When all Vmds in a Device's Mdib have an ApprovedJurisdictions
+    //       list, then this test case should not be applied to it.
```
Comment (ldeichmann) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java:174]: This is definitely not the right place to "inform" anyone when not to apply the test, no user will ever read it.

**Review by ldeichmann — CHANGES_REQUESTED** [review]

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java (hunk 1 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java]: Reference removed.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java (hunk 2 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java]: Reference removed.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java (hunk 3 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java]: Adapted the note. Please check again.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java (hunk 4 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantBicepsNormativeAnnexTest.java]: Made sure this is also stated in RQM.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java (hunk 5 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java:126]: Made sure this is also stated in RQM.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java (hunk 6 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantParticipantModelAnnexTest.java:174]: Made sure this is also stated in RQM.

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java [hunk 7]
```diff
@@ -341,6 +342,7 @@ void testRequirementR0013() throws NoTestData, IOException {
         assertTestData(acceptableSequenceSeen.get(), "No suitable context states seen, test failed.");
     }
 
+    // TODO: Add a Precondition
```
[the remark this answers is not in the document]
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantNonFunctionalQualityAttributesTest.java]: Removed the TODO alltogether.

**Review by ldeichmann — APPROVED** [review]

**Pull request merged**